### PR TITLE
Add auth during cloning

### LINKED ISSUE #46: gopages - auth is not passed to git.clone - fails on private repos
Hi, I have created an orphaned `gh-pages` branch, and have working PAT, which I tested by cloning with it:

```
> git clone https://$USER:$PAT@github.com/org/repo.git 
Cloning into 'repo'...
remote: Enumerating objects: 531, done.
remote: Counting objects: 100% (282/282), done.
remote: Compressing objects: 100% (154/154), done.
remote: Total 531 (delta 168), reused 156 (delta 109), pack-reused 249
Receiving objects: 100% (531/531), 316.76 KiB | 3.23 MiB/s, done.
Resolving deltas: 100% (241/241), done.
```

I can check out that branch:
```
(main> > git checkout gh-pages
(gh-pages) > git pull origin gh-pages
From https://github.com/org/repo
 * branch            gh-pages   -> FETCH_HEAD
Already up to date.
```

However when trying to run:

```
 gopages \ 
                -base /go/ \
                -out "/Users/user/repo/" \
                -gh-pages \
                -gh-pages-user "$USER" \
                -gh-pages-token "$PAT"

```

I am getting:

```
pipe: Failed to clone in-memory copy of repo. Be sure the 'gh-pages' orphaned branch exists: ... : authentication required
```

What am I missing?..

### PR DESCRIPTION
**Pull request by shlomi-dr**
Otherwise, this doesn't work for private repos

fixes #46

### CODE REVIEW

File: gopages/main.go [hunk 1]
```diff
@@ -114,10 +114,15 @@ var generateMemfsDocsPipe = pipe.New(pipe.Options{}).
 		return args[0].(memfsDocsArgs)
 	}).
 	Append(func(args memfsDocsArgs) (memfsDocsArgs, *git.Repository, error) {
+		var auth *gitHTTP.BasicAuth = nil
+		if args.Flags.GitHubPagesUser != "" || args.Flags.GitHubPagesToken != "" {
+			auth = &gitHTTP.BasicAuth{Username: args.Flags.GitHubPagesUser, Password: args.Flags.GitHubPagesToken}
+		}
```
Comment (JohnStarich) [inline gopages/main.go:120]: This change looks good and works well locally.

No pressure, but would you mind adding a regression test to cover this scenario?
One easy option might be to extract this bit to a separate func, then add a unit test for both zero auth and GitHub token auth.

Comment (shlomi-dr) [reply gopages/main.go]: Added tests as asked. Hope this is good now! :)

**Review by JohnStarich — APPROVED** [review]
Looks great, thanks @shlomi-dr!

**Comment by shlomi-dr** [comment]
Thank you for making this great tool, and opening it for us to use 😊
